src/helpers/users.js

```diff
@@ -110,7 +111,18 @@ exports.getUser = async (userId, credentials) => {
   return user;
 };
 
-exports.userExists = async email => User.findOne({ 'local.email': email }).lean();
+exports.userExists = async (email, credentials) => {
+  const user = await User.findOne({ 'local.email': email }).lean();
+
+  const targetUserCompany = user ? user.company.toHexString() : '';
+  const loggedUserCompany = credentials.company ? credentials.company._id.toHexString() : '';
+  const hasVendorRole = has(credentials, 'role.vendor');
+  const isFromCompany = loggedUserCompany === targetUserCompany;
+
+  return hasVendorRole || isFromCompany
```

Comment:
je ne suis pas sur de comprendre la raison de ces conditions

Reply:
ausssi pour le nom de tes variables : dans les faits `has` et `is` ne dirige pas vers la meme personne et ce n'est pas ce qui forcement clair

Reply:
done